typing/types.mli

```diff
@@ -588,14 +582,6 @@ and mixed_product_shape =
     flat_suffix : flat_element array;
   }
 
-||||||| 121bedcfd2
-=======
-and type_origin =
-    Definition
-  | Rec_check_regularity       (* See Typedecl.transl_type_decl *)
-  | Existential of string
-
->>>>>>> 5.2.0
```

Comment:
I suspect we should take the upstream side here. The `abstract_reason` and `type_origin` types are mostly the same, and we should pick the one that reduces drift with upstream.